src/sonic-host-services/scripts/hostcfgd

```diff
@@ -514,119 +501,134 @@ class HostConfigDaemon:
+        elif cached_feature_state in ("always_enabled", "always_disabled"):
+            disable = state == "always_disabled"
+            enable = state == "always_enabled"
+        elif cached_feature_state in ("enabled", "disabled"):
+            enable = state == "enabled"
+            disable = state == "disabled"
+        else:
+            syslog.syslog(syslog.LOG_INFO, "Feature {} service is {}".format(feature_name, cached_feature_state))
+            return False
 
-    def update_all_feature_states(self):
+        if not enable and not disable:
+            syslog.syslog(syslog.LOG_ERR, "Unexpected state value '{}' for feature {}"
+                          .format(state, feature_name))
+            return False
+
+        feature_names, feature_suffixes = self.get_feature_attribute(feature_name, feature_cfg)
+
+        if enable:
+            self.enable_feature(feature_names, feature_suffixes)
+            syslog.syslog(syslog.LOG_INFO, "Feature {} is enabled and started".format(feature_name))
+
+        if disable:
+            self.disable_feature(feature_names, feature_suffixes)
+            syslog.syslog(syslog.LOG_INFO, "Feature {} is stopped and disabled".format(feature_name))
+
+        return True
+
+    def render_all_feature_states(self):
+        """
+        Render the Template (if any) for the state field of the FEATURE Table.
+        Update the rendered state in the config db
+        """
         feature_table = self.config_db.get_table('FEATURE')
         for feature_name in feature_table:
             if not feature_name:
                 syslog.syslog(syslog.LOG_WARNING, "Feature is None")
                 continue
 
-            state = feature_table[feature_name]['state']
+            state = feature_table.get(feature_name, {}).get('state', '')
             if not state:
                 syslog.syslog(syslog.LOG_WARNING, "Enable state of feature '{}' is None".format(feature_name))
                 continue
 
-            target_state = self.get_target_state(feature_name, state)
-            # Store the initial value of 'state' field in 'FEATURE' table of a specific container
-            self.cached_feature_states[feature_name] = target_state
+            self.set_target_state(feature_name, state)
+    
+    def set_target_state(self, feature_name, state):
+        template = jinja2.Template(state)
+        target_state = template.render(self.device_config)
+        entry = self.config_db.get_entry('FEATURE', feature_name)
+        entry["state"] = target_state
+        self.config_db.set_entry("FEATURE", feature_name, entry)
 
-            self.update_feature_state(feature_name, target_state, feature_table)
+    def __get_intf_name(self, key):
+        if isinstance(key, tuple) and key:
+            intf = key[0]
+        else:
+            intf = key
+        return intf
 
-    def aaa_handler(self, key, data):
+    def aaa_handler(self, key, op, data):
         self.aaacfg.aaa_update(key, data)
```

Comment:
>aaa_update [](http://example.com/codeflow?start=20&length=10)

You missed a syslog statement, comparing with [https://github.com/Azure/sonic-buildimage/pull/8861](https://github.com/Azure/sonic-buildimage/pull/8861)